models/utility_functions.py

```diff
@@ -16,6 +16,34 @@
 from ttnn.device import Arch
 
 
+def find_mismatched_indices_and_values(tensor1, tensor2, rtol, atol):
+    if tensor1.shape != tensor2.shape:
+        raise ValueError("Tensors must have the same shape")
+
+    # compute difference tensor
+    difference = torch.abs(tensor1 - tensor2) > atol + rtol * torch.abs(tensor2)
+
+    # find diff idx
+    mismatched_indices = torch.nonzero(difference, as_tuple=False)
+
+    # get different value
+    mismatched_values = [
+        (idx.tolist(), tensor1[tuple(idx)].item(), tensor2[tuple(idx)].item()) for idx in mismatched_indices
+    ]
+
+    return mismatched_indices, mismatched_values
+
+
+def print_mismatched_elements(golden, calculated, rtol, atol):
+    indicies, values = find_mismatched_indices_and_values(golden, calculated, rtol=rtol, atol=atol)
+
+    MAX_PRINT_CNT = 10
```

Comment:
If `len(indices) > MAX_PRINT_CNT`, then you should still only display 10 different idx/indices, however perhaps you could add further logging to say that more indices are mismatched and only `MAX_PRINT_CNT` has been printed.

Reply:
![image](https://github.com/user-attachments/assets/2b789cff-ddd4-4b64-abbd-769c0e9d3f4b)

I’ve updated the message. Now, if the number exceeds `MAX_PRINT_CNT`, an additional message will appear below.
`logger.debug("More indices are mismatched and only MAX_PRINT_CNT={} Displayed.", MAX_PRINT_CNT)`